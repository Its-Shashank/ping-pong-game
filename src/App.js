import BeforeStart from "./containers/BeforeStart/Loadable";
import GameStart from "./containers/StartGame/Loadable";
import styled from "styled-components";
import React, { useState } from "react";

const CustomContainer = styled.div`
  height: 100vh;
  display: flex;
  align-items: center;
`;

function App() {
  const [startGame, setStartGame] = useState(false);
  const handleGameStart = () => {
    setStartGame(!startGame);
  };
  return (
    <CustomContainer>
      {!startGame ? <BeforeStart handleGameStart={handleGameStart} /> : <GameStart handleGameStart={handleGameStart} />}
    </CustomContainer>
  );
}

export default App;
