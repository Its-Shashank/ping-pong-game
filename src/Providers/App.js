import React, { useState } from "react";
import BeforeStart from "./containers/BeforeStart/Loadable";
import GameStart from "./containers/StartGame/Loadable";
import { gameCreators } from "./reducer";
import makeSelectGame, { selectPlayers, selectPlayerOneScore, selectPlayerTwoScore } from "./selectors";
import styled from "styled-components";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { compose } from "redux";
import { createStructuredSelector } from "reselect";

const CustomContainer = styled.div`
  height: 100vh;
  display: flex;
  align-items: center;
`;

function App({
  players,
  playerOneScore,
  playerTwoScore,
  dispatchResetScore,
  dispatchSetPlayers,
  dispatchSetPlayerOneScore,
  dispatchSetPlayerTwoScore
}) {
  const [startGame, setStartGame] = useState(false);
  const [player1, setPlayer1] = useState(players.playerOne);
  const [player2, setPlayer2] = useState(players.playerTwo);
  const [error, setError] = useState(false);
  const handleGameStart = () => {
    if (player2 !== "" && player1 !== "") {
      setError(false);
      setStartGame(!startGame);
      dispatchSetPlayers({ player1, player2 });
    } else {
      setError(true);
    }
  };
  const toggleGameStart = () => {
    setStartGame(!startGame);
  };
  const handlePlayerOne = (e) => {
    setError(false);
    setPlayer1(e);
  };
  const handlePlayerTwo = (e) => {
    setError(false);
    setPlayer2(e);
  };
  const incrementPlayerOneScore = () => {
    console.log({ playerOneScore, playerTwoScore, players });
    dispatchSetPlayerOneScore(playerOneScore + 1);
  };
  const incrementPlayerTwoScore = () => {
    dispatchSetPlayerTwoScore(playerTwoScore + 1);
  };
  const handleReset = () => {
    dispatchResetScore();
  };
  return (
    <CustomContainer>
      {!startGame ? (
        <BeforeStart
          handleGameStart={handleGameStart}
          handlePlayerOne={handlePlayerOne}
          handlePlayerTwo={handlePlayerTwo}
          player1={player1}
          player2={player2}
          error={error}
        />
      ) : (
        <GameStart
          handleGameStart={toggleGameStart}
          player1={player1}
          player2={player2}
          incrementPlayerOneScore={incrementPlayerOneScore}
          incrementPlayerTwoScore={incrementPlayerTwoScore}
          playerOneScore={playerOneScore}
          playerTwoScore={playerTwoScore}
          handleReset={handleReset}
        />
      )}
    </CustomContainer>
  );
}
App.propTypes = {
  dispatchResetScore: PropTypes.func,
  dispatchSetPlayers: PropTypes.func,
  dispatchSetPlayerOneScore: PropTypes.func,
  dispatchSetPlayerTwoScore: PropTypes.func,
  playerOneScore: PropTypes.number,
  playerTwoScore: PropTypes.number,
  players: PropTypes.object
};

const mapStateToProps = createStructuredSelector({
  game: makeSelectGame(),
  players: selectPlayers(),
  playerOneScore: selectPlayerOneScore(),
  playerTwoScore: selectPlayerTwoScore()
});

function mapDispatchToProps(dispatch) {
  return {
    dispatchSetPlayerOneScore: (score) => dispatch(gameCreators.setPlayerOneScore(score)),
    dispatchSetPlayerTwoScore: (score) => dispatch(gameCreators.setPlayerTwoScore(score)),
    dispatchSetPlayers: (names) => dispatch(gameCreators.setPlayers(names)),
    dispatchResetScore: () => dispatch(gameCreators.resetScore()),
    dispatchClearPlayersData: () => dispatch(gameCreators.clearPlayersData())
  };
}
const withConnect = connect(mapStateToProps, mapDispatchToProps);
export default compose(withConnect)(App);
export const AppTest = compose()(App);
