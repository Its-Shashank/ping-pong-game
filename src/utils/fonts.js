export const FONT_SIZES = {
  player_names: "30px",
  win_text: "20px",
  score: "60px"
};
