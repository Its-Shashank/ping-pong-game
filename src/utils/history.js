import { createBrowserHistory } from "history";
const baseUrl = process.env.NODE_ENV === "production" ? "/ping-pong" : "/";
const history = createBrowserHistory({ basename: baseUrl });
export default history;
