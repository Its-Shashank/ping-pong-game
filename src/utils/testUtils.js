import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { browserHistory } from "react-router-dom";
import configureStore from "../configureStore";

export const renderProvider = (children) => {
  const store = configureStore({}, browserHistory).store;
  return render(<Provider store={store}>{children}</Provider>);
};
