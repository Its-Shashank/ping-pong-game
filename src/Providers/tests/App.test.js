/* eslint-disable no-undef */
/**
 *
 * Tests for Keeper
 *
 *
 */

import React from "react";
import { renderProvider } from "../../utils/testUtils";
import { AppTest as Game } from "../App";

describe("<Game Keeper /> container tests", () => {
  beforeEach(() => {});
  it("should render and match the snapshot", () => {
    const { baseElement } = renderProvider(<Game />);
    expect(baseElement).toMatchSnapshot();
  });
});
