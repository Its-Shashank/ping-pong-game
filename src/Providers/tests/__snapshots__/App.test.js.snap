// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game Keeper /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ jwNwTN"
    />
  </div>
</body>
`;
